### DISCUSSION TITLE
How to do a RaceGaps/Cummulative delta plot like this one?

### DISCUSSION THREAD

**Discussion by yagomosquera** [theOehrly/Fast-F1#301] [Q&A]
<img width="750" alt="Captura de pantalla 2022-12-13 a las 17 46 44" src="https://user-images.githubusercontent.com/120037523/207393765-0672e91c-7712-4112-9347-70868e90f9b3.png">

I tried to use the utils module, but there was no way to finish something like this.

**Comment by theOehrly** [comment]
What were you using for this? `utils.delta_time`? That function should at most be used for the delta within one lap, not for an entire race. And even then the results need to be treated with caution.
For a plot like the one you want to create, you should use the "Time" at which each lap was set, available in `Session.laps`. Then, you can calculate the distance to other drivers at the end of each lap.
Alternatively, `api.timing_data` has a so called 'stream_data' return value which gives the time difference to the car ahead in frequent intervals. You could also use this data.
In both cases, you will need to do the calculations for this kind of plot yourself.

**Reply by yagomosquera** [reply]
actually tried to do it, but something is not working properly as at the end of the race my results are not the same as the actual race, hehe. 

```
lapss= session.laps

drivers = lapss['Driver'].unique()

result_dfs = []

for driver in drivers:
    laps = lapss[lapss['Driver'] == driver]
    laps.iloc[0, laps.columns.get_loc('LapTime')] = laps["LapStartTime"].iat[1]-laps["LapStartTime"].iat[0]
    laps['LapTimeSeconds'] = laps['LapTime'].dt.total_seconds()
    baseline = laps['LapTimeSeconds'].quantile(0.25)
    laps['DeltaSeconds'] = baseline - laps['LapTimeSeconds']
    laps['AcuDeltaSeconds'] = laps['DeltaSeconds'].cumsum()
    result_dfs.append(laps)
result_dfs = pd.concat(result_dfs)
```

My guess is that everything messes up after de safety car... I also re-used this code as it calculates the time of the 1º lap and I can extract the positions gained-lost on 1º lap.

**Reply by theOehrly** [reply]
Have you searched the discussions on here? I think someone else was trying to do the same thing and shared some code. But I'm not sure.
